core/services/workflows/syncer/handler.go

```diff
@@ -156,17 +157,22 @@ func NewEventHandler(
 	lggr logger.Logger,
 	workflowStore store.Store,
 	capRegistry core.CapabilitiesRegistry,
+	engineRegistry *EngineRegistry,
 	emitter custmsg.MessageEmitter,
 	ratelimiter *ratelimiter.RateLimiter,
 	workflowLimits *syncerlimiter.Limits,
 	workflowArtifacts WorkflowArtifactsStore,
 	opts ...func(*eventHandler),
-) *eventHandler {
+) (*eventHandler, error) {
+	if engineRegistry == nil {
```

Comment:
nil check should come after applying the options

Reply:
https://github.com/smartcontractkit/chainlink/pull/16925#discussion_r2037573037

Reply:
Argument for why it's okay is that to override the engineRegistry using an opts function you would need to add the Opts func. So if that change ever happens then this check should move then.